nixos/modules/services/display-managers/sddm.nix

```
@@ -66,7 +66,7 @@ let
       HideShells = "/run/current-system/sw/bin/nologin";
     };
 
-    X11 = {
+    X11 = optionalAttrs config.services.xserver.enable {
```

```suggestion
    X11 = optionalAttrs xcfg.enable {
```

I had this change in my testing branch but probably forgot to carry it over to the PR.